### ISSUE TITLE
brew linkapps not picking up Frescobaldi 2.18.1?

### ISSUE THREAD

**Issue by AllanKinnaird** [Homebrew/homebrew#40139]
After brew update and brew upgrade --all to install Frescobaldi 2.18.1, brew linkapps doesn't produce an alias in Applications. The command seems to be working OK with other apps.

**Comment by mistydemeo** [comment]
This version of frescobaldi doesn't appear to be installed as an application bundle; it's unclear to me whether this change was intentional, though the PR that updated it doesn't seem to have made any specific changes to that effect.

**Comment by AllanKinnaird** [comment]
Previously it has operated a normal application, so I'd guess the change is unintentional. See the caveat in the formula.

**Comment by DomT4** [comment]
@mistydemeo https://github.com/Homebrew/homebrew/pull/39107#issuecomment-97444666

**Comment by DomT4** [comment]
> See the caveat in the formula.

Apologies. The caveat should have been cut as well - I didn't read it thoroughly enough; I presumed it impacted the executable that launches an "app" via Python as well.

**Comment by AllanKinnaird** [comment]
I guess I don’t understand the system as well as I should. I thought the caveat was eminently sensible and helpful. Can we keep the caveat notice and still have it treated as a bundled app?

**Comment by DomT4** [comment]
>  Can we keep the caveat notice and still have it treated as a bundled app?

If you can get @mikemcquaid's :+1: I'll revert the removal of the app bundle. The issues raised around ML and app creation problems in the PR linked above remain true, but I'm not against a revert if you can convince the other maintainers.

**Comment by AllanKinnaird** [comment]
I’m totally in agreement with the ideal ((not idea!!) of having only one version of an installation to cover all OS versions, but we have to accept that sometimes Apple make changes which cause incompatibilities moving upwards through their system versions. I have a PPC machine which I keep really only to play Myst and Chess Master.

I may be wrong but my belief is that the vast majority of users who are sufficiently interested in their computer’s functionality that they find and use Homebrew, will tend to keep their system fully updated. Hence  the wish to update Frescobaldi to the latest available.

Reading the posts, it seems that the problem is the installation of the latest version of Frescobaldi on OS X 10.8. Would people be installing the very latest Frescobaldi  on an out-of-date OS?

According to Netmarketshare, users of 10.8 currently account for 5.8% of the OS X total. Of those, how many are likely to be using Frescobaldi provided through Homebrew? (Does github by any chance maintain statistics on os versions calling for downloads?)

It seems to me that to reduce the functionality of the installation for all users results in a greater negative consequence than having the very much smaller number of 10.8 users stay with a version that works for them. I hope you can reconsider the removal of the bundle.

**Comment by mikemcquaid** [comment]
The real issue is that the previous `.app` was not officially provided/supported by upstream and thus should never really have been added. When it broke we tried to debug it, failed and had to remove the functionality.

Homebrew is not good at managing `.app` bundles and we try to avoid doing so, as a result. It doesn't provide a good user experience and people blame us when things don't work rather than the upstream project which should port their software to OS X properly and provide a downloadable `.app` binary (ideally with an auto-updater too).
